.github/workflows/sonar-qube.yml

```diff
@@ -0,0 +1,38 @@
+name: SonarCloud Code Analysis
+
+on:
+    pull_request:
+        types:
+            - opened
+            - synchronize
+            - reopened
+    push:
+        branches:
+            - main
+            - next
+
+jobs:
+    sonarqube:
+        name: SonarQube
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  fetch-depth: 0
+
+            - uses: pnpm/action-setup@v4
+
+            - name: Use Node.js 22.x
+              uses: actions/setup-node@v4
+              with:
+                  node-version: 22
+                  registry-url: "https://registry.npmjs.org"
+                  cache: "pnpm"
+
+            - run: pnpm install --frozen-lockfile
+
```

Comment:
Is this necessary? Shouldn't Sonar work without dependencies?

https://github.com/SonarSource/sonarcloud-github-action?tab=readme-ov-file#usage

Reply:
that is needed for the tsconfig files there are imports from packages inside the extends